zerver/lib/test_runner.py

```diff
@@ -472,9 +486,25 @@ def run_tests(self, test_labels: List[str],
         result = self.run_suite(suite)
         self.teardown_test_environment()
         failed = self.suite_result(suite, result)
+        failed_thread = []  # type: List[str]
         if not failed:
             write_instrumentation_reports(full_suite=full_suite, include_webhooks=include_webhooks)
-        return failed, result.failed_tests
+        else:
+            # We check which thread a failed test ran in by looking in each
+            # worker's log.  When the test is found, load the contents of the file.
+            failed_thread_path = "var/test-backend-*.json"
+            try:
+                for filename in glob.glob(failed_thread_path):
+                    with open(filename, 'r') as f:
+                        for line in f:
+                            if result.failed_tests[0] in line:
```

Comment:
I'm not entirely sure if it's necessary to iterate through this list or not?  Or, I guess the question is, how should multiple failures be handled?